qutip/entropy.py

```diff
@@ -244,24 +243,42 @@ def _entropy_relative(rho, sigma, base=e, sparse=False):
     rel_ent : float
         Value of relative entropy.
 
+    References
+    ----------
+
+    See Nielsen & Chuang, "Quantum Computation and Quantum Information",
+    Section 11.3.1, pg. 511 for a detailed explanation of quantum relative
+    entropy.
     """
-    if rho.type != 'oper' or sigma.type != 'oper':
-        raise TypeError("Inputs must be density matrices..")
-    # sigma terms
-    svals = sp_eigs(sigma.data, sigma.isherm, vecs=False, sparse=sparse)
-    snzvals = svals[svals != 0]
+    if not rho.isoper or not sigma.isoper:
+        raise TypeError("Inputs must be density matrices.")
```

Comment:
Oh yeah, one more point - I think elsewhere in QuTiP we promote kets to dms in functions like this.  The relative entropy is defined between states, and a ket has an unambiguous representation as a dm without losing information, so we can probably implement that.

Reply:
Good idea. Implemented.